hibernate-core/src/test/java/org/hibernate/orm/test/query/QuerySelectWhereArrayPositionTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.query;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import org.hibernate.dialect.PostgreSQLDialect;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.RequiresDialect;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@DomainModel(
+		annotatedClasses = {
+				QuerySelectWhereArrayPositionTest.TestEntity.class,
+		}
+)
+@SessionFactory
+@RequiresDialect( PostgreSQLDialect.class )
+@JiraKey("HHH-19490")
+public class QuerySelectWhereArrayPositionTest {
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					session.persist(new TestEntity(List.of(1, 2, 3)));
+					session.persist(new TestEntity(List.of(2, 3, 4)));
+					session.persist(new TestEntity(List.of(3, 4, 5)));
+				}
+		);
+	}
+
+	@Test
+	public void testSelectWhereArrayPosition(SessionFactoryScope scope) {
+		scope.inTransaction( session -> {
+			List<TestEntity> result = session.createQuery( "select testEntity "
+														+ "from TestEntity testEntity "
+														+ "where ARRAY_POSITION(testEntity.values, ?1) IS NOT NULL", TestEntity.class )
+					.setParameter( 1, 3)
+					.getResultList();
+			assertThat( result.size() ).isEqualTo( 3 );
```

Comment:
Perhaps make one of the persisted entities not have an element at position `3`, so we also ensure the predicate works as expected.